extensions/npm/src/commands.ts

```diff
@@ -28,4 +32,29 @@ export function runSelectedScript() {
 		let message = localize('noScriptFound', 'Could not find a valid npm script at the selection.');
 		vscode.window.showErrorMessage(message);
 	}
-}
\ No newline at end of file
+}
+
+export async function selectAndRunScriptFromFolder(folderInfo: vscode.Uri) {
+	type TaskMap = { [id: string]: vscode.Task; };
+	let taskList = await detectNpmScriptsForFolder(folderInfo.path);
+
+	if (taskList && taskList.length > 0) {
+		let taskMap: TaskMap = {};
+		taskList.forEach(t => {
+			let uri = getPackageJsonUriFromTask(t);
+			if (uri && uri.fsPath.length >= folderInfo.fsPath.length) {
+				let taskName = uri.fsPath.substring(folderInfo.fsPath.length, uri.fsPath.length - '/package.json'.length) + ' > ' + t.name.substring(0, t.name.search('-'));
+				taskMap[taskName] = t;
+			}
+		});
+		let result = await vscode.window.showQuickPick(Object.keys(taskMap).sort(), {
+			placeHolder: `Run scripts on folder ${folderInfo.fsPath}...`,
+		});
+		if (result) {
+			vscode.tasks.executeTask(taskMap[result]);
+		}
+	}
+	else {
+		vscode.window.showInformationMessage(`No scripts detected in folder ${folderInfo.path}`);
```

Comment:
This should be a modal dialog and not a notification. The message should be "No npm scripts found in ..."